### PR TITLE
[fix](memory) Fix LRU cache deleter and memory tracking

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

In order to add common code to the value deleter of LRU cache, let all lru cache values inherit from LRUCacheValueBase class and tracking memory in destructor.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/runtime/memory/lru_cache_policy.h [hunk 1]
```diff
@@ -72,7 +81,48 @@ class LRUCachePolicy : public CachePolicy {
...
+    }
+
+    std::shared_ptr<MemTrackerLimiter> mem_tracker() { return _mem_tracker; }
+
+    Cache::Handle* insert(const CacheKey& key, void* value, size_t charge,
+                          CachePriority priority = CachePriority::NORMAL, size_t bytes = -1) {
+        return _insert(true, key, value, charge, priority, bytes);
+    }
+
+    Cache::Handle* insert_no_tracking(const CacheKey& key, void* value, size_t charge,
```
Comment (yiguolei) [inline be/src/runtime/memory/lru_cache_policy.h:102]: 删了吧，直接size = 0 就是no tracking了

Comment (xinyiZzz) [reply be/src/runtime/memory/lru_cache_policy.h:102]: done

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/runtime/memory/lru_cache_policy.h [hunk 2]
```diff
@@ -72,7 +81,54 @@ class LRUCachePolicy : public CachePolicy {
...
+                          CachePriority priority = CachePriority::NORMAL) {
+        size_t bytes_with_handle = _get_bytes_with_handle(key, charge, tracking_bytes);
+        if (tracking_bytes > 0) {
+            _mem_tracker->cache_consume(bytes_with_handle);
+        }
+        DorisMetrics::instance()->lru_cache_memory_bytes->increment(bytes_with_handle);
+        if (value != nullptr) {
+            ((LRUCacheValueBase*)value)
+                    ->bind_memory_tracking(bytes_with_handle,
+                                           tracking_bytes > 0 ? _mem_tracker.get() : nullptr);
```
[the remark this answers is not in the document]
Comment (xinyiZzz) [reply be/src/runtime/memory/lru_cache_policy.h]: done

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**